### DISCUSSION TITLE
i have error in kali linux

### DISCUSSION THREAD

**Discussion by karim0811** [moby/moby#47400] [Show and tell]
sudo add-apt-repository ppa:kelebek333/kablosuz
Repository: 'deb https://ppa.launchpadcontent.net/kelebek333/kablosuz/ubuntu/ kali-rolling main'
Description:
PPA for Realtek wifi drivers

If you installed your system in UEFI mode, you must disable secureboot for modules installed by dkms after installation.

You can check installation mode with following command.

[ -d /sys/firmware/efi ] && echo "EFI" || echo "BIOS"

You can check status of secureboot with following command.

mokutil --sb-state
More info: https://launchpad.net/~kelebek333/+archive/ubuntu/kablosuz
Adding repository.
Press [ENTER] to continue or Ctrl-c to cancel.
Found existing deb entry in /etc/apt/sources.list.d/kelebek333-ubuntu-kablosuz-kali-rolling.list
Adding deb entry to /etc/apt/sources.list.d/kelebek333-ubuntu-kablosuz-kali-rolling.list
Found existing deb-src entry in /etc/apt/sources.list.d/kelebek333-ubuntu-kablosuz-kali-rolling.list
Adding disabled deb-src entry to /etc/apt/sources.list.d/kelebek333-ubuntu-kablosuz-kali-rolling.list
Adding key to /etc/apt/trusted.gpg.d/kelebek333-ubuntu-kablosuz.gpg with fingerprint F30D3DBFB7534F670B44C71923E50C670722A6D9
Ign:1 https://apt.dockerproject.org/repo ubuntu-xenial InRelease
Ign:2 https://apt.dockerproject.org/repo ubuntu-trusty InRelease                                                                           
Hit:3 http://kali.download/kali kali-bleeding-edge InRelease                                                                               
Hit:4 http://kali.download/kali kali-experimental InRelease                                                                                  
Ign:5 https://download.docker.com/linux/debian kali-rolling InRelease                                                                        
Hit:6 https://download.docker.com/linux/debian bullseye InRelease                      
Hit:7 http://kali.download/kali kali-rolling InRelease                            
Ign:1 https://apt.dockerproject.org/repo ubuntu-xenial InRelease                  
Ign:2 https://apt.dockerproject.org/repo ubuntu-trusty InRelease                 
Err:8 https://download.docker.com/linux/debian kali-rolling Release              
  404  Not Found [IP: 108.158.245.53 443]
Ign:1 https://apt.dockerproject.org/repo ubuntu-xenial InRelease                 
Ign:2 https://apt.dockerproject.org/repo ubuntu-trusty InRelease
Ign:9 https://ppa.launchpadcontent.net/kelebek333/kablosuz/ubuntu kali-rolling InRelease
Err:10 https://ppa.launchpadcontent.net/kelebek333/kablosuz/ubuntu kali-rolling Release
  404  Not Found [IP: 185.125.190.80 443]
Err:1 https://apt.dockerproject.org/repo ubuntu-xenial InRelease
  Something wicked happened resolving 'apt.dockerproject.org:https' (-5 - No address associated with hostname)
Err:2 https://apt.dockerproject.org/repo ubuntu-trusty InRelease
  Something wicked happened resolving 'apt.dockerproject.org:https' (-5 - No address associated with hostname)
Reading package lists... Done
W: https://download.docker.com/linux/debian/dists/bullseye/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
E: The repository 'https://download.docker.com/linux/debian kali-rolling Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'https://ppa.launchpadcontent.net/kelebek333/kablosuz/ubuntu kali-rolling Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

**Comment by thaJeztah** [comment]
It looks like you have the apt package repository set up incorrectly for your distro.


I see something is trying to reference the obsolete `apt.dockerproject.org` package repositories for Ubuntu Xenial (16.04) and Ubuntu Trusty (14.04). The `apt.dockerproject.org` repository as a whole is obsolete for a long time, and both Ubuntu 14.04 and 16.04 reached EOL many Years ago;

```
Ign:1 https://apt.dockerproject.org/repo ubuntu-xenial InRelease
Ign:2 https://apt.dockerproject.org/repo ubuntu-trusty InRelease
```

It also tries to use the current package repository at `download.docker.com`, but trying to install Ubuntu packages, using `kali-rolling` as distro version, which won't match any of the Ubuntu versions it provides packages for;

```
E: The repository 'https://download.docker.com/linux/debian kali-rolling Release' does not have a Release file.
```

Note that download.docker.com does _NOT_ provide packages for Kali Linux. Kali Linux is a Debian derivative, and based on "Debian testing" (unstable); https://www.kali.org/docs/policy/kali-linux-relationship-with-debian/

> The Kali Linux distribution is based on Debian Testing. Therefore, most of the Kali packages are imported, as-is, from the Debian repositories. In some cases, newer packages may be imported from Debian Unstable or Debian Experimental, either to improve user experience, or to incorporate needed bug fixes.

As it is a Debian derivative, you  _may_ be able to install packages using the [installation instructions for Debian](https://docs.docker.com/engine/install/debian/), but when setting up the package repository, you need to substitute the distro-version with the version that matches the Debian equivslent that your version of Kali Linux was based on https://docs.docker.com/engine/install/#other-linux-distros. However, as those packages were created for Debian, they've not been tested on Kali Linux.
